examples/pipe/pipe_main.c

```diff
@@ -204,26 +206,59 @@ int main(int argc, FAR char *argv[])
 
   if (ret != 0)
     {
-      fprintf(stderr, "pipe_main: PIPE test FAILED (%d)\n", ret);
+      fprintf(stderr, "pipe_main: pipe() transfer_test test FAILED "
+              "(%d)\n", ret);
       return 9;
     }
 
-  /* Perform the pipe redirection test */
+  fprintf(stderr, "pipe_main: pipe() transfer_test test PASSED\n");
+
+  /* Perform the pipe() redirection test */
 
-  printf("\npipe_main: Performing redirection test\n");
+  fprintf(stderr, "\npipe_main: Performing pipe() redirection test\n");
   ret = redirection_test();
   if (ret != 0)
     {
-      fprintf(stderr, "pipe_main: PIPE redirection test FAILED (%d)\n", ret);
+      fprintf(stderr, "pipe_main: pipe() redirection test FAILED (%d)\n",
+              ret);
       return 10;
     }
 
-  printf("pipe_main: PIPE redirection test PASSED\n");
+  fprintf(stderr, "pipe_main: pipe() redirection test PASSED\n");
+
+  /* Test file_pipe() command */
+
+  fprintf(stderr, "\npipe_main: Open with file_pipe()\n");
+
+  filepipep[0] = &filepipe[0];
+  filepipep[1] = &filepipe[1];
+
+  ret = file_pipe(filepipep, CONFIG_DEV_PIPE_SIZE, 0);
```

Comment:
it isn't good to call kernel specific function from user space. let's test pipe or open_pty instead.

Reply:
You are right!

We'd need to test using pty, but it doesn't seem to fit on this test. I'll drop it for now.